crates/client/src/server.rs

```diff
@@ -0,0 +1,87 @@
+/// Server abstraction module.
+pub mod commands;
+
+mod errors;
+
+pub use commands::Command;
+pub use errors::Result;
+
+use crate::args::Args;
+use clap::Parser;
+use io::{Read, Write};
+use std::{io, net::TcpStream};
+
+use self::commands::Msg;
+
+/// Abstraction over the server.
+#[allow(dead_code)]
+#[derive(Debug)]
+pub struct Server {
+    /// The stream to the server.
+    stream: TcpStream,
+    /// The width of the map.
+    width: usize,
+    /// The height of the map.
+    height: usize,
+    /// Read buffer.
+    buf: String,
+}
+
+impl Server {
+    /// Creates a new server instance and connects to it.
+    pub fn new() -> Result<Self> {
+        let args = Args::parse();
+        let stream = TcpStream::connect((args.host.as_str(), args.port))?;
+        let mut self_ = Self {
+            stream,
+            width: 0,
```

Comment:
I hope this is not some kind of anti-pattern? I did it like this to use the `get_line` method, but I can do it in some other manner.